packages/url-state-provider/src/index.ts

```diff
@@ -152,16 +168,15 @@ function updateState(stateID, state, options = {}) {
  * changes to the method.
  * @param {boolean} historyOptions.replace - If true it replaces the last state in the window history (default false).
  */
-function updateStateAndHistory(stateID, state, merge, historyOptions) {
-  historyOptions = historyOptions || {}
+function updateStateAndHistory(stateID: string, state: object, merge: boolean, historyOptions: HistoryOptions = {}) {
   const newUrl = updateState(stateID, state, { merge })
   const historyState = historyOptions.state || ""
   const historyTitle = historyOptions.title || ""
 
   if (historyOptions?.replace) {
-    window.history.replaceState(historyState, historyTitle, newUrl)
+    window.history.replaceState(historyState, historyTitle, newUrl as string)
```

Comment:
avoid casting modify origin of method to return string

Reply:
From how I understand andreas methods it is, in fact a object. Which is somehow saved as a string.